packages/router/src/router.ts

```diff
@@ -1475,7 +1476,9 @@ function validateCommands(commands: string[]): void {
   for (let i = 0; i < commands.length; i++) {
     const cmd = commands[i];
     if (cmd == null) {
-      throw new Error(`The requested path contains ${cmd} segment at index ${i}`);
+      throw new RuntimeError(
+          RuntimeErrorCode.NULLISH_COMMAND,
+          `The requested path contains ${cmd} segment at index ${i}`);
```

Comment:
nit: this could use `ngDevMode` check if the entire function is not guarded yet:
```suggestion
          NG_DEV_MODE &&  `The requested path contains ${cmd} segment at index ${i}`);
```